toolchain/check/diagnostic_helpers.h

```diff
@@ -18,15 +18,33 @@ namespace Carbon::Check {
 // explicitly construct a `LocId` from it first.
 class LocIdForDiagnostics {
  public:
+  // Constructs a token-only location for a diagnostic.
+  //
+  // This means the displayed location will include only the location's specific
+  // parse node, instead of also including its descendants. As such, this can
+  // only be true for locations backed by a `NodeId`.
```

Comment:
It wasn't clear to me which cases "backed by a `NodeId`" might cover without reading the implementation, but it looks like the only case this is excluding is a location imported from C++. Would it be worth mentioning that here? Maybe:
```suggestion
  // parse node, instead of also including its descendants. This has no effect
  // for locations that refer to C++ source files.
```
(We could make `token_only` mode highlight the token in C++ diagnostics too, so maybe even this restriction could be lifted eventually.)

Reply:
I'll adjust the comment, but I think we end up with no associated LocId in other situations too, like implicit imports.

Reply:
Maybe there's something I'm missing. As far as I can see, the only consumer of this type is `ConvertLoc`, and every path through `ConvertLoc` ends up in `ConvertLocInFile`, and the only way that doesn't have a node ID to combine with the `token_only` flag is when the location is a C++ source location. For a location in an import, it looks like we step through the imports producing the "in import" diagnostics and then apply the `token_only` flag to the token in the imported file.

Reply:
Switched. I was mixing up with desugaring in my head, sorry.

Not actually used by C++ right now.